pkg/fieldpath/fieldpath.go

```diff
@@ -57,3 +69,29 @@ func ExtractFieldPathAsString(obj interface{}, fieldPath string) (string, error)
 
 	return "", fmt.Errorf("unsupported fieldPath: %v", fieldPath)
 }
+
+// SplitMaybeSubscriptedPath checks whether the specified fieldPath is
+// subscripted, and
+//  - if yes, this function splits the fieldPath into path and subscript, and
+//    returns (path, subscript, true).
+//  - if no, this function returns (fieldPath, "", false).
+//
+// Example inputs and outputs:
+//  - "metadata.annotations['myKey']" --> ("metadata.annotations", "myKey", true)
+//  - "metadata.annotations['a[b]c']" --> ("metadata.annotations", "a[b]c", true)
+//  - "metadata.labels"               --> ("metadata.labels", "", false)
+//  - "metadata.labels['']"           --> ("metadata.labels", "", false)
+func SplitMaybeSubscriptedPath(fieldPath string) (string, string, bool) {
+	if !strings.HasSuffix(fieldPath, "']") {
```

Comment:
I'm afraid to say it, but what about whitespace?  Do we allow `annotations[ 'foobar' ]` ?

Reply:
It's not allowed. I feel like we could just require `['` and `']` to be the identifier/delimiter of the key.